### PR TITLE
Replace NumericInput control with NumberInput control

### PR DESCRIPTION
**Pull request by bsteinbk**
The NumericInput control was a wrapper around react-numeric-control. Because of bugs uncovered in this base control it was determined to just replace the control with one written using the latest React technology which makes it much easier to maintain in the future. This PR also marks the old NumericInput control as deprecated.

### CODE REVIEW

File: ui/core/src/ui-core.ts [hunk 1]
```diff
@@ -78,6 +78,7 @@ export * from "./ui-core/inputs/LabeledInput";
 export * from "./ui-core/inputs/LabeledTextarea";
 export * from "./ui-core/inputs/numericinput/NumericInput";
 export * from "./ui-core/inputs/numericinput/ReactNumericInput";
+export * from "./ui-core/inputs/numericinput/NumberInput";
```
Comment (DanEastBentley) [inline ui/core/src/ui-core.ts:81]: Should we create a new subdirectory named "numberinput"?

File: ui/core/src/ui-core/inputs/numericinput/NumberInput.scss [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+@import "../../base/base";
+
+.core-numeric-input-container {
```
Comment (DanEastBentley) [inline ui/core/src/ui-core/inputs/numericinput/NumberInput.scss:7]: ".core-number-input" should probably be used here instead.

File: ui/core/src/ui-core.ts (hunk 1 above)
Comment (bsteinbk) [reply ui/core/src/ui-core.ts]: done

File: ui/core/src/ui-core/inputs/numericinput/NumberInput.scss (hunk 2 above)
Comment (bsteinbk) [reply ui/core/src/ui-core/inputs/numericinput/NumberInput.scss]: Renamed all occurrences.

**Review by DanEastBentley — APPROVED** [review]

**Review by wgoehrig — DISMISSED** [review]
@bsteinbk This PR has whitespace-only changes caused by a (recently-fixed) bug in our commit hooks.

Please run the following locally to revert these changes:

```
git co master
git pull
git co <your branch>
git merge master
rush fix-copyrights
```

**Review by GerardasB — APPROVED** [review]

**Review by pmconne — APPROVED** [review]

**Pull request merged**